api-metastore/src/main/java/org/zalando/nakadi/service/AvroSchemaCompatibility.java

```diff
@@ -0,0 +1,153 @@
+package org.zalando.nakadi.service;
+
+import org.apache.avro.AvroRuntimeException;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaCompatibility;
+import org.apache.avro.SchemaCompatibility.Incompatibility;
+import org.apache.avro.SchemaCompatibility.SchemaPairCompatibility;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.domain.CompatibilityMode;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class AvroSchemaCompatibility {
+
+    private static BiFunction<Schema, Schema, SchemaPairCompatibility> backwardCompatibleFn = (reader, writer) ->
+            SchemaCompatibility.checkReaderWriterCompatibility(reader, writer);
```

Comment:
is avro library thread safe? I expect shema validation can happen from different threads

Reply:
I could not find any document stating that `SchemaCompatibility.checkReaderWriterCompatibility` is thread-safe but going through the code of this method I didnt find any state that would be shared for different calls to this method. Every call results in creation of new objects.